packages/vega-spec-builder/src/bar/barSpecBuilder.ts

```diff
@@ -166,6 +192,10 @@ export const addData = produce<Data[], [BarSpecOptions]>((data, options) => {
 	if (type === 'dodged' || isDodgedAndStacked(options)) {
 		data[index].transform?.push(getDodgeGroupTransform(options));
 	}
+
+	if (isDualMetricAxis(options)) {
+		addDualMetricAxisData(data, options);
+	}
```

Comment:
Can we move the if statement into the `addDualMetricAxisData()` function instead of here? Similar to what we do for `addTendlineData`, `addTooltipData`, and `addPopoverData`